app/components/primer/octicon_symbols_component.rb

```diff
@@ -13,7 +13,26 @@ class OcticonSymbolsComponent < Primer::Component
     #
     # @param icons [Array<Hash>] List of icons to render, in the format { symbol: :icon_name }
     def initialize(icons: [])
-      @icons = icons
+      @icons = {}
+      icons.each do |icon|
+        symbol = icon[:symbol]
+        size = Primer::OcticonComponent::SIZE_MAPPINGS[
+          fetch_or_fallback(Primer::OcticonComponent::SIZE_OPTIONS, icon[:size] || Primer::OcticonComponent::SIZE_DEFAULT, Primer::OcticonComponent::SIZE_DEFAULT)
+        ]
+
+        cache_key = Primer::Octicon::Cache.get_key(symbol: symbol, size: size)
+
+        if (cache_icon = Primer::Octicon::Cache.read(cache_key))
+          icon_instance = cache_icon
+        else
+          icon_instance = Octicons::Octicon.new(symbol, height: size)
+
+          Primer::Octicon::Cache.set(cache_key, icon_instance)
+        end
+
+        # Don't put the same icon twice
+        @icons[[symbol, icon_instance.height]] = icon_instance if @icons[[symbol, icon_instance.height]].nil?
```

Comment:
is `icon_instance.height == size`?

if so, we could add a 
`next if @icons[[symbol, size]].present?` before reading the cache

Reply:
Not always. If someone asks for a medium size and the octicon gem doesn't have a 24px size, it will return the 16px octicon. In that case size would equal medium and height would equal 16